src/main/java/at/hannibal2/skyhanni/features/mining/ActualGemstonePowderDisplay.kt

```diff
@@ -0,0 +1,285 @@
+import at.hannibal2.skyhanni.utils.chat.TextHelper
+import at.hannibal2.skyhanni.utils.compat.hover
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+
+@SkyHanniModule
+object ActualGemstonePowderDisplay {
+
+    private data class MultiplierBreakdown(
+        val atomizedCrystals: Double,
+        val echoOfAtomizedRate: Double,
+        val is2xEventActive: Boolean,
+        val isSkymallActive: Boolean,
+        val hotmBuffPercent: Int,
+        val drillMultiplier: Double,
+        val petMultiplier: Double,
+        val petName: String?,
+        val totalMultiplier: Double
+    )
+
+
+    private val config get() = SkyHanniMod.feature.chat
+
+    private var is2xPowderActive = false
+
+    private val patternGroup = RepoPattern.group("mining.powder.multiplier")
+    private val powderBossBarPattern by patternGroup.pattern(
+        "powder.bossbar",
+        "§e§lPASSIVE EVENT §b§l2X POWDER §e§lRUNNING FOR §a§l(?<time>.*)§r",
+    )
+
+    private val drillBasePowderPattern by patternGroup.pattern(
+        "drill.powder.base",
+        "§7§7Grants §d\\+(?<bonus>\\d+)% §dGemstone Powder§7, and"
+    )
+
+    private val drillUpgradePowderPattern by patternGroup.pattern(
+        "drill.powder.upgrade",
+        "§7Earn §9\\+(?<bonus>\\d+)% Powder §7from all sources."
+    )
+
+
+    private fun getAtomizedCrystalsBonus(): Double {
+        return 0.10 // 10%
+    }
+
+    private fun getEchoOfAtomizedBonusRate(): Double {
+        return 0.02 // 2%
+    }
+
+
+    @HandleEvent
+    fun onSecondPassed() {
+        if (!isEnabled()) return
+
+        is2xPowderActive = false
+
+        if (TabWidget.EVENT.isActive) {
+            for (line in TabWidget.EVENT.lines) {
+                if (line.contains("2x Powder")) {
+                    is2xPowderActive = true
+                    break
+                }
+            }
+        } else {
+            powderBossBarPattern.matchMatcher(BossbarData.getBossbar()) {
+                is2xPowderActive = true
+            }
+        }
+    }
+
+
+    private fun calculateMultiplierBreakdown(): MultiplierBreakdown {
+
+        var additiveBonus = 0.0
+
+        val atomizedCrystalsMulti = getAtomizedCrystalsBonus()
+        val echoOfAtomizedRate = getEchoOfAtomizedBonusRate()
+
+        additiveBonus += atomizedCrystalsMulti
+
+        val echoContribution = atomizedCrystalsMulti * echoOfAtomizedRate
+        additiveBonus += echoContribution
+
+        var totalMultiplier = 1.0 + additiveBonus
+
+        val is2xActive = is2xPowderActive
+        if (is2xActive) {
+            totalMultiplier *= 2.0
+        }
+
+        val isSkymallActive = HotmData.SKY_MALL.enabled && HotmApi.skymall == HotmApi.SkymallPerk.EXTRA_POWDER
+        if (isSkymallActive) {
+            totalMultiplier *= 1.15
+        }
+
+
+
+        val hotmBuffPercent = if (HotmData.POWDER_BUFF.enabled) {
+            HotmData.POWDER_BUFF.activeLevel
+        } else 0
```

Comment:
The activeLevel already does the enable check intern and returns 0 otherwise.
```suggestion
        val hotmBuffPercent = HotmData.POWDER_BUFF.activeLevel
```